go/arrow/array/numeric.gen.go

```diff
@@ -750,6 +962,22 @@ func (a *Timestamp) setData(data *Data) {
 	}
 }
 
+func (a *Timestamp) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+	return a.values[i].ToTime(a.DataType().(*arrow.TimestampType).Unit).Format("2006-01-02 15:04:05.999999999")
```

Comment:
should formatting be conditional on precision?

Reply:
using `9` there will leave out trailing zeros from the resulting string so this will handle whatever level of precision is provided rather than explicitly padding to a specific precision level.